src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -3,47 +3,110 @@
-    private static final Logger LOGGER = Logger.getLogger(CustomLogger.class.getName());
     private static final Multiset<String> messageCount = HashMultiset.create();
     private static final String LOGGER_LAYOUT = "%d | %-5p| %c{1} | %m%n";
+    // log4j2 uses SYSTEM_OUT and SYSTEM_ERR - support both
+    private static final String SYSTEM_OUT_ALT = "STDOUT";
+    private static final String SYSTEM_ERR_ALT = "STDERR";
 
     /** Sets up the custom logger to the specified level and location. */
     public static void setup(Level level, String logLocation) {
+        final LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+        final Configuration config = ctx.getConfiguration();
+        String target = "CONSOLE";
+
         if (logLocation != null
-                && !ConsoleAppender.SYSTEM_ERR.equals(logLocation)
-                && !ConsoleAppender.SYSTEM_OUT.equals(logLocation)) {
-            RollingFileAppender fa = new RollingFileAppender();
-            fa.setName("FileLogger");
-            fa.setFile(logLocation);
-            fa.setMaxFileSize("5MB");
-            fa.setMaxBackupIndex(1);
-            fa.setLayout(new PatternLayout(LOGGER_LAYOUT));
-            fa.setThreshold(level);
-            fa.setAppend(true);
-            fa.activateOptions();
-            Logger.getRootLogger().addAppender(fa);
-            LOGGER.info("File Handler set");
+                && !ConsoleAppender.Target.SYSTEM_ERR.toString().equals(logLocation)
+                && !SYSTEM_ERR_ALT.equals(logLocation)
+                && !ConsoleAppender.Target.SYSTEM_OUT.toString().equals(logLocation)
+                && !SYSTEM_OUT_ALT.equals(logLocation)) {
+
+            target = "FileLogger";
+
+            PatternLayout layout = PatternLayout.newBuilder()
+                .withConfiguration(config)
+                .withPattern(LOGGER_LAYOUT)
+                .build();
+
+            RollingFileAppender fa = RollingFileAppender.newBuilder()
+                .setConfiguration(config)
+                .withName(target)
+                .withLayout(layout)
+                .withFileName(logLocation)
+                .withFilePattern(logLocation + ".%d")
+                .withPolicy(SizeBasedTriggeringPolicy.createPolicy("5MB"))
+                .withStrategy(DefaultRolloverStrategy.newBuilder().withMax("1").build())
+                .build();
+
+            fa.start();
+            config.addAppender(fa);
+            ctx.getRootLogger().addAppender(config.getAppender(fa.getName()));
+
+            log.info("File Handler set");
         } else {
-            ConsoleAppender consoleAppender = new ConsoleAppender(new PatternLayout(LOGGER_LAYOUT));
-            if (logLocation != null) {
-                consoleAppender.setTarget(logLocation);
+
+            if (logLocation != null
+                    && (ConsoleAppender.Target.SYSTEM_ERR.toString().equals(logLocation)
+                        || SYSTEM_ERR_ALT.equals(logLocation))) {
```

Comment:
Shouldn't we allow `SYSTEM_OUT`/ `SYSTEM_OUT_ALT` along with  `SYSTEM_ERR`/ `SYSTEM_ERR_ALT`?

Reply:
Maybe.... but this is actually the same on `master`, most of these changes came in from a cherry-pick. I had to fix a few things, but this is really the same everywhere. I don't oppose doing some follow-up and maybe verifying/fixing a couple of things in the logger logic, but this should be 👍 and consistent with our logging on JMXFetch on most recent versions.

Thanks for the review @prognant 🙇